sdk/include/debug.hh

```diff
@@ -741,4 +741,89 @@ namespace
 		Assert(auto, const char *, Ts &&...) -> Assert<Ts...>;
 	};
 
+	enum class StackCheckMode
+	{
+		Disabled,
+		Logging,
+		Asserting,
+	};
+
+	/**
+	 * Check the (dynamic) stack usage of a function, including all of its
+	 * callees.  This is intended to be used in compartment entry points to
+	 * check the stack size that is required for that entry point.  Use this
+	 * with some test vectors that explore different code paths to identify the
+	 * maximum stack usage.  This can then be used with the
+	 * [[cheriot::minimum_stack]] attribute to ensure that the switcher will
+	 * never invoke the entry point with insufficient stack.
+	 *
+	 * Note that this records only the stack usage for the current compartment
+	 * invocation (including any invoked shared libraries).  If this
+	 * compartment calls others, then smaller stacks may fail.  This failure is
```

Comment:
I'm not quite sure what this sentence is saying.  Do you mean "If this compartment calls others, then even stacks larger than this recorded threshold may fail"?

Reply:
Clarified.